### ISSUE TITLE
Hooks to run on commited/changed files only

### ISSUE THREAD

**Issue by T4rk1n** [plotly/dash#2542]
Hooks takes a long time to run (~1minute) on every commit, we could run the linting on the files that are committed/changed only to save time.

**Comment by alexcjohnson** [comment]
Good idea. I end up using `git commit --no-verify` frequently, when I'm confident that the linters won't find anything, but it's annoying to have to think about that.
